src/include/shared/vio_event_channel.h

```diff
@@ -18,7 +18,6 @@ typedef struct enc_dev_config
 {
     uint8_t dev_id;
     enc_evt_channel_t* enc_evt_chn;
```

Comment:
Pointer chain here is unnecessarily complex. uint64_t enclave_evt_channel, uint64_t host_evt_channel, uint32_t qidx all shoud be in the share memory. Defining enc_dec_config as below should simplify the memory check in enclave.
`````
{
    uint8_t dev_id;
    evt_t *enc_evt_channel;
    evt_t *host_evt_channel;
    uint32_t *qidx_p
}
```

Reply:
This brings a lot of complications in other parts of the code (e.g. during setup on the host side and `host_evt_channel`  and `enclave_evt_channel` would have to be a `evt_t**`) and I'm not sure those parts are even correct. @prp @davidchisnall: is it intended and correct that `host_dev_event_channel_init` ([here](https://github.com/lsds/sgx-lkl/blob/fcd488160888ef111508cbf1865c3d59dcbdd83f/src/main-oe/sgxlkl_evt_chn_cfg.c#L44)) allocates `evt_channel_num` many `enc_evt_channel_t`s for each event channel with number `evt_channel_num`? Are these essentially (n^2)/2 communication links?